src/main/java/menion/android/whereyougo/utils/NotificationService.java

```diff
@@ -0,0 +1,123 @@
+package menion.android.whereyougo.utils;
+
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.PendingIntent;
+import android.app.Service;
+import android.app.NotificationManager;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Build;
+import android.os.IBinder;
+import android.support.v4.app.NotificationCompat;
+
+public class NotificationService extends Service {
+    private static final int notification_id = 10;
+    private static final String NOTIFICATION_CHANNEL_ID = "menion.android.whereyougo.utils.NotificationService";
+    private boolean foreground = false;
+    private boolean runing = false;
+    private NotificationManager mNM;
+    private String contentTitel;
+
+    public static final String TITEL = "ContentTitel";
+    public static final String START_NOTIFICATION_SERVICE = "START_NOTIFICATION_SERVICE";
+    public static final String START_NOTIFICATION_SERVICE_FOREGROUND = "START_NOTIFICATION_SERVICE_FOREGROUND";
+    public static final String STOP_NOTIFICATION_SERVICE = "STOP_NOTIFICATION_SERVICE";
+
+    public NotificationService() {
+    }
+
+    @Override
+    public IBinder onBind(Intent intent) {
+        // Used only in case if service are bound (Bound Services).
+        return null;
+    }
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+        Logger.v(this.getClass().getName(), "onCreate()");
```

Comment:
Please replace this.getClass().getName() by a static definition.
Hardcoding the class name is ugly but the only reliable way to work with proguard.

Reply:
I have replaced this by a static definition.